Cabal/Distribution/Simple/Build.hs

```diff
@@ -406,8 +407,9 @@ testSuiteLibV09AsLibAndExe pkg_descr
     libClbi = LibComponentLocalBuildInfo
                 { componentPackageDeps = componentPackageDeps clbi
                 , componentPackageRenaming = componentPackageRenaming clbi
-                , componentLibraries = [LibraryName (testName test)]
+                , componentLibraryName = LibraryName "test"
```

Comment:
I tried to construct a failing test-case for this, but the detailed-0.9 doesn't publish a test library, so you can't actually use the two test suites together. Still, it costs little to fix. I'll update the V09 to also test multiple libraries.